locksmith/src/operations/userOperations.ts

```diff
@@ -0,0 +1,47 @@
+const ethJsUtil = require('ethereumjs-util')
+const models = require('../models')
+const { User, UserReference } = models
+import * as types from '../types'
+
+namespace UserOperations {
+  export async function createUser(
+    input: types.UserCreationInput
+  ): Promise<Boolean> {
+    let userReference = await UserReference.create(
+      {
+        emailAddress: input.emailAddress.toLowerCase(),
+        User: {
+          publicKey: ethJsUtil.toChecksumAddress(input.publicKey),
+          recoveryPhrase: input.recoveryPhrase,
+          passwordEncryptedPrivateKey: input.passwordEncryptedPrivateKey,
+        },
+      },
+      {
+        include: User,
+      }
+    )
+
+    if (userReference) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  export async function getUserPrivateKeyByEmailAddress(
+    emailAddress: String
+  ): Promise<String | null> {
+    let user = await UserReference.findOne({
+      where: { emailAddress: emailAddress.toLowerCase() },
+      include: [{ model: User, attributes: ['passwordEncryptedPrivateKey'] }],
+    })
+
+    if (user) {
+      return user.User.passwordEncryptedPrivateKey
+    } else {
+      return null
```

Comment:
That's where we should generate a private key and encrypt it with a salted hash of the email address

Reply:
This branch of the conditional represents the encrypted private key not existing in storage. From our last conversation, I believe that we were/are not generating private keys at all on the back end and this falls under the concern of the front end